### PR TITLE
test(amplify-provider-awscloudformation): add tests for amplify-resource-state-utils.ts

### PR DESCRIPTION
**Pull request by rajrajhans**
#### Description of changes

add tests for  `amplify-resource-state-utils.ts` in `amplify-provider-awscloudformation` package.

#### Issue #, if available

#7050 

#### Description of how you validated changes
`yarn test`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const expectedTableNameMap: Map<string, string> = new Map();
+    expectedTableNameMap.set('LogicalResourceIdTest1', 'TestStackOutputValue1');
+
+    const tableNames = await getTableNames(cfnClientStub, tables, StackID);
+    expect(tableNames).toEqual(expectedTableNameMap);
+  });
+
+  test('test getPreviousDeploymentRecord', async () => {
+    const expectedPrevDeploymentRecord = {
+      'capabilities': ['CAPABILITY_IAM'],
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts:54]:
```suggestion
      capabilities: ['CAPABILITY_IAM'],
```

File: packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      'parameters': { 'TestParameterKey1': 'TestParameterValue1' },
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts:55]:
```suggestion
      parameters: { TestParameterKey1: 'TestParameterValue1' },
```

**Review by cjihrig — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { getTableNames, getPreviousDeploymentRecord } from '../../utils/amplify-resource-state-utils';
+import { CloudFormation } from 'aws-sdk';
+
+const cfnClientStub = ({
+  describeStackResources: ({ StackName }) => ({
+    promise: () => ({
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts:6]: I think this mock, and the one a few lines down should use `Promise.resolve()`.

Comment (rajrajhans) [reply packages/amplify-provider-awscloudformation/src/__tests__/utils/amplify-resource-state-utils.test.ts]: Thanks. Made the required changes 👍🏻

**Pull request merged**